src/compiler/utilities.ts

```diff
@@ -1504,7 +1529,7 @@ namespace ts {
     }
 
     // True if the given identifier, string literal, or number literal is the name of a declaration node
-    export function isDeclarationName(name: Node): name is Identifier | StringLiteral | LiteralExpression {
+    export function isDeclarationName(name: Node): boolean {

```

Comment:
Probably because this function can also return `true` if the node is some identifier, but not used as a declaration. With the control flow checks, the code after an `if` with a return will be considered to be the else block, which might have caused some issues.